sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilterSuite.scala

```diff
@@ -527,4 +538,54 @@ class ParquetFilterSuite extends QueryTest with ParquetTest with SharedSQLContex
       assert(df.filter("_1 IS NOT NULL").count() === 4)
     }
   }
+
+  test("Fiters should be pushed down for vectorized Parquet reader at row group level") {
+    import testImplicits._
+    withTempPath { dir =>
+      val path = s"${dir.getCanonicalPath}/table"
+      (1 to 1024).map(i => (101, i)).toDF("a", "b").write.parquet(path)
+
+      val requiredSchema = StructType(Seq(
+        StructField("a", IntegerType, nullable = false),
+        StructField("b", IntegerType, nullable = false)
+      ))
+
+      val hadoopConf = new Configuration()
+
+      // Set up parquet block size to make sure many row groups produced.
+      hadoopConf.setInt("parquet.block.size", 8)
+      hadoopConf.set(ParquetInputFormat.READ_SUPPORT_CLASS, classOf[ParquetReadSupport].getName)
+      hadoopConf.set(
+        ParquetReadSupport.SPARK_ROW_REQUESTED_SCHEMA,
+        ParquetSchemaConverter.checkFieldNames(requiredSchema).json)
+
+      val filters = ParquetFilters.createFilter(requiredSchema, sources.LessThan("a", 100))
+      assert(filters.isDefined)
+
+      val attemptId = new TaskAttemptID(new TaskID(new JobID(), TaskType.MAP, 0), 0)
+      val hadoopAttemptContext1 =
+        new TaskAttemptContextImpl(hadoopConf, attemptId)
+      val hadoopAttemptContext2 =
+        new TaskAttemptContextImpl(hadoopConf, attemptId)
+
+      ParquetInputFormat.setFilterPredicate(hadoopAttemptContext1.getConfiguration, filters.get)
```

Comment:
I think we should test the ParquetInputFormat to do the push down, otherwise we are testing parquet itself.

In order to known how many row groups are read, we could have a SQL metrics for that.